Shared/Samples/Show viewshed from point on map/ShowViewshedFromPointOnMapView.swift

```diff
@@ -0,0 +1,202 @@
+        
+        /// The graphics overlay that displays the viewshed for the tapped location.
+        let resultGraphicsOverlay: GraphicsOverlay = {
+            let resultGraphicsOverlay = GraphicsOverlay()
+            let fillColor = UIColor.orange.withAlphaComponent(0.4)
+            let fillSymbol = SimpleFillSymbol(
+                style: .solid,
+                color: fillColor
+            )
+            let resultRenderer = SimpleRenderer(symbol: fillSymbol)
+            resultGraphicsOverlay.renderer = resultRenderer
+            return resultGraphicsOverlay
+        }()
+        
+        /// The graphics overlay that shows where the user tapped on the screen.
+        let inputGraphicsOverlay: GraphicsOverlay = {
+            let inputGraphicsOverlay = GraphicsOverlay()
+            // Red dot marker that is added to graphic overlay on map on the location where the user taps.
+            let pointSymbol = SimpleMarkerSymbol(style: .circle, color: .red, size: 10)
+            // Sets the renderer to draw the dot symbol on the graphics overlay.
+            let renderer = SimpleRenderer(symbol: pointSymbol)
+            inputGraphicsOverlay.renderer = renderer
+            return inputGraphicsOverlay
+        }()
+        
+        /// Removes previously tapped location from overlay and draws new dot on tap location.
+        /// - Parameter point: Location to add a dot graphic on the map.
+        func addInputGraphic(at point: Point) {
+            inputGraphicsOverlay.removeAllGraphics()
+            let graphic = Graphic(geometry: point)
+            inputGraphicsOverlay.addGraphic(graphic)
+        }
+        
+        /// Controls the initialization for the geoprocessing of the viewshed and calls the geoprocessing logic.
+        /// - Parameter point: Location that the user tapped on the map.
+        func calculateViewshed(at point: Point) async throws {
+            // Clears previously viewshed drawing.
+            resultGraphicsOverlay.removeAllGraphics()
+            // If there is a geoprocessing job in progress it is cancelled.
+            await geoprocessingJob?.cancel()
+            guard let spatialReference = point.spatialReference else { return }
+            // Creates a feature collection table based on the spatial reference for the tapped location
+            // on the map.
+            let featureCollectionTable = FeatureCollectionTable(
+                fields: [Field](),
+                geometryType: Point.self,
+                spatialReference: spatialReference
+            )
+            do {
+                // Creates a feature for the point tapped.
+                let feature = featureCollectionTable.makeFeature(geometry: point)
+                // Asynchronously adds that feature to the table.
+                try await featureCollectionTable.add(feature)
+                try await performGeoprocessing(featureCollectionTable)
+            } catch {
+                throw error
+            }
```

Comment:
If a method is throwing, no need to do-catch then throw the error again. Same for other methods below.

```suggestion
            // Creates a feature for the point tapped.
            let feature = featureCollectionTable.makeFeature(geometry: point)
            // Asynchronously adds that feature to the table.
            try await featureCollectionTable.add(feature)
            try await performGeoprocessing(featureCollectionTable)
```